acvm-repo/brillig_vm/src/arithmetic.rs

```diff
@@ -70,46 +68,127 @@ pub(crate) fn evaluate_binary_int_op<F: AcirField>(
     rhs: MemoryValue<F>,
     bit_size: IntegerBitSize,
 ) -> Result<MemoryValue<F>, BrilligArithmeticError> {
-    let lhs = lhs.expect_integer_with_bit_size(bit_size).map_err(|err| match err {
-        MemoryTypeError::MismatchedBitSize { value_bit_size, expected_bit_size } => {
-            BrilligArithmeticError::MismatchedLhsBitSize {
-                lhs_bit_size: value_bit_size,
-                op_bit_size: expected_bit_size,
+    if matches!(
+        op,
+        BinaryIntOp::Add
+            | BinaryIntOp::Sub
+            | BinaryIntOp::Mul
+            | BinaryIntOp::Div
+            | BinaryIntOp::And
+            | BinaryIntOp::Or
+            | BinaryIntOp::Xor
+    ) {
+        match (lhs, rhs, bit_size) {
+            (MemoryValue::U1(lhs), MemoryValue::U1(rhs), IntegerBitSize::U1) => {
+                evaluate_binary_int_op_u1(op, lhs, rhs).map(MemoryValue::U1)
```

Comment:
`evaluate_binary_int_op_u1` doesn't handle `Sub`, `Mul` and `Add`, just `Div`, is that okay?

Reply:
We do handle these cases, `Add` and `Sub` are equivalent to `Xor` and `Mul` is equivalent to `And` so I've just merged those branches.

Reply:
Ah, right, I didn't see 👍